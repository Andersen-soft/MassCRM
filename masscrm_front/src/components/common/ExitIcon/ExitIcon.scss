.icon {
  width: 18px;
  height: 18px;
  fill: #dadada;

  &_active {
    cursor: pointer;

    &:hover {
      fill: #69738f;
    }
  }
}
