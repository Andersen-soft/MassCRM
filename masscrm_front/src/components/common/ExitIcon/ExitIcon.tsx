import React, { FC } from 'react';
import { styleNames } from 'src/services';
import { IExitIconProps } from './interfaces';
import styles from './ExitIcon.scss';

const sn = styleNames(styles);

export const ExitIcon: FC<IExitIconProps> = ({ onClickHandler }) => {
  const icon = <use xlinkHref='assets/svg/sprite.svg#exit' />;

  return onClickHandler ? (
    <span
      id='logout-button'
      onClick={onClickHandler}
      role='button'
      tabIndex={0}
      onKeyDown={
        onClickHandler &&
        (button => {
          button.keyCode === 13 && button.key === 'Enter' && onClickHandler();
        })
      }
    >
      <svg className={`${sn('icon')} ${!!onClickHandler && sn('icon_active')}`}>
        {icon}
      </svg>
    </span>
  ) : (
    <span>
      <svg className={sn('icon')}>{icon}</svg>
    </span>
  );
};
