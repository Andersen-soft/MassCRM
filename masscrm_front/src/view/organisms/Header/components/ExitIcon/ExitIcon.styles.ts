import { makeStyles } from '@material-ui/core';

export const useStyles = makeStyles({
  icon: {
    width: '18px',
    height: '18px',
    fill: '#dadada'
  },
  active: {
    cursor: 'pointer',
    '&:hover': {
      fill: '#69738f'
    }
  }
});
