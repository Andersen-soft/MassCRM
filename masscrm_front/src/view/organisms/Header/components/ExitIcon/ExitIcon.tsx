import React, { FC } from 'react';
import { KEY_ENTER } from 'src/constants';
import { useStyles } from './ExitIcon.styles';

interface IProps {
  onClickHandler?: () => void;
}

export const ExitIcon: FC<IProps> = ({ onClickHandler }) => {
  const styles = useStyles();
  const icon = <use xlinkHref='assets/svg/sprite.svg#exit' />;

  return onClickHandler ? (
    <span
      id='logout-button'
      onClick={onClickHandler}
      role='button'
      tabIndex={0}
      onKeyDown={
        onClickHandler &&
        (button => {
          button.key === KEY_ENTER.keyName && onClickHandler();
        })
      }
    >
      <svg className={`${styles.icon} ${!!onClickHandler && styles.active}`}>
        {icon}
      </svg>
    </span>
  ) : (
    <span>
      <svg className={styles.icon}>{icon}</svg>
    </span>
  );
};
