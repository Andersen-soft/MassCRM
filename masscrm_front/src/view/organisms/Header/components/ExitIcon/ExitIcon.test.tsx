import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { ExitIcon } from './ExitIcon';

describe('ExitIcon', () => {
  const mockCallBack = jest.fn();

  const icon = shallow(<ExitIcon onClickHandler={mockCallBack} />);

  it('Logo should render correctly', () => {
    expect(shallowToJson(icon)).toMatchSnapshot();
  });

  icon.simulate('click');
  expect(mockCallBack.mock.calls.length).toEqual(1);
});
