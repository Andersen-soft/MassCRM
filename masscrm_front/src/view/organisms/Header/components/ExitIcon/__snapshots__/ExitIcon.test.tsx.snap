// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExitIcon Logo should render correctly 1`] = `
<span
  id="logout-button"
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  onKeyDown={[Function]}
  role="button"
  tabIndex={0}
>
  <svg
    className="makeStyles-icon-1 makeStyles-active-2"
  >
    <use
      xlinkHref="assets/svg/sprite.svg#exit"
    />
  </svg>
</span>
`;
